vault/request_handling.go

```diff
@@ -439,7 +439,12 @@ func (c *Core) switchedLockHandleRequest(httpCtx context.Context, req *logical.R
 		return nil, fmt.Errorf("could not parse namespace from http context: %w", err)
 	}
 
-	resp, err = c.handleCancelableRequest(namespace.ContextWithNamespace(ctx, ns), req)
+	ctx = namespace.ContextWithNamespace(ctx, ns)
+	inFlightReqID, ok := httpCtx.Value("in-flight-reqID").(string)
```

Comment:
There's a nice pattern I like to use for WithValue where you define an empty type to use as the key to ensure conflicts can't occur, see `ctxKeyForwardedRequestMountAccessor` for an example.